### PR TITLE
Testing with Vagrant

### PR DESCRIPTION
**Pull request by TomaszGaweda**
PR closed by Hazelcast automation as no activity (>3 months). Please reopen with comments, if necessary. Thank you for using Hazelcast and your valuable contributions

### CODE REVIEW

File: .github/workflows/publish-packages.yml [hunk 1]
```diff
@@ -323,5 +307,136 @@ jobs:
 
       - name: Uninstall Hazelcast from homebrew
         run: |
-          source ./common.sh          
+          source ./common.sh
           brew uninstall ${{ env.HZ_DISTRIBUTION}}@$BREW_PACKAGE_VERSION
+
+  buildVagrant:
+    runs-on: macos-12
+    needs: [ prepare, rpm, deb ]
```
Comment (ldziedziul) [inline .github/workflows/publish-packages.yml:315]: Do you need `prepare` here?

File: .github/workflows/publish-packages.yml [hunk 2]
```diff
@@ -323,5 +307,136 @@ jobs:
...
+        run: |
+          source ./common.sh
+          vagrant up
+
+      - name: Run Vagrant halt
+        run: vagrant halt
+
+  testOnVagrant:
+    runs-on: macos-12
+    needs: [prepare, buildVagrant]
```
Comment (ldziedziul) [inline .github/workflows/publish-packages.yml:352]: Do you need `prepare` here?

File: .github/workflows/publish-packages.yml [hunk 3]
```diff
@@ -323,5 +307,136 @@ jobs:
...
+      HZ_VERSION: ${{ needs.prepare.outputs.hz_version }}
+      PACKAGE_VERSION: ${{ needs.prepare.outputs.package_version }}
+
+    steps:
+      - name: Checkout hazelcast-packaging repo
+        uses: actions/checkout@v2.3.2
+        with:
+          path: 'hazelcast-packaging'
+
+      - name: Cache Vagrant boxes
```
Comment (ldziedziul) [inline .github/workflows/publish-packages.yml:378]: You already have this step in `buildVagrant`. Is it needed here?

File: .github/workflows/publish-packages.yml [hunk 4]
```diff
@@ -323,5 +307,136 @@ jobs:
...
+      matrix:
+        distribution: [ 'hazelcast', 'hazelcast-enterprise' ]
+    env:
+      HZ_VERSION: ${{ needs.prepare.outputs.hz_version }}
+      PACKAGE_VERSION: ${{ needs.prepare.outputs.package_version }}
+      HZ_DISTRIBUTION: ${{ matrix.distribution }}
+    defaults:
+      run:
+        working-directory: ./hazelcast-packaging
+    needs: [ prepare, testOnVagrant ]
```
Comment (ldziedziul) [inline .github/workflows/publish-packages.yml:421]: Same here about `prepare`

File: check-hazelcast-health.sh [hunk 5]
```diff
@@ -6,7 +6,7 @@ until $(curl --silent --fail "127.0.0.1:5701/hazelcast/health/ready"); do
   if [ ${attempts} -eq ${max_attempts} ];then
       echo "Hazelcast not responding"
       cat hz.log
-      ps -aux
+      ps aux
```
Comment (ldziedziul) [inline check-hazelcast-health.sh:9]: Why?

File: .github/workflows/publish-packages.yml [hunk 6]
```diff
@@ -323,5 +307,136 @@ jobs:
...
+      - name: Run Hazelcast ${{ matrix.hz_distribution }} on ${{ matrix.linux_distribution }}
+        run: |
+          vagrant ssh ${{ env.LINUX_DISTRIBUTION}} -c \
+            'HAZELCAST_CONFIG="/home/vagrant/hz-config.yaml" hz-start > ~/hz.log 2>&1 &'
+
+      - name: Wait for cluster to be up
+        run: |
+          vagrant ssh ${{ env.LINUX_DISTRIBUTION}} -c '~/check-hazelcast-health.sh'
+
+      - name: check health via script
```
Comment (ldziedziul) [inline .github/workflows/publish-packages.yml:400]: Could we check systemd functionality?

See description of the following PRs:
- https://github.com/hazelcast/hazelcast-packaging/pull/120
- https://github.com/hazelcast/hazelcast-packaging/pull/121

**Review by ldziedziul — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-packages.yml (hunk 1 above)
Comment (TomaszGaweda) [reply .github/workflows/publish-packages.yml:315]: For the `HZ_VERSION: ${{ needs.prepare.outputs.hz_version }}`

File: .github/workflows/publish-packages.yml (hunk 4 above)
Comment (TomaszGaweda) [reply .github/workflows/publish-packages.yml:421]: For the `HZ_VERSION: ${{ needs.prepare.outputs.hz_version }}` as everywhere

File: check-hazelcast-health.sh (hunk 5 above)
Comment (TomaszGaweda) [reply check-hazelcast-health.sh:9]: Doesn't work on Mac (`no user named x`)

File: .github/workflows/publish-packages.yml (hunk 6 above)
Comment (TomaszGaweda) [reply .github/workflows/publish-packages.yml:400]: b70824b070b37746ca9a691e0cae80b9550c0921

Comment (ldziedziul) [reply .github/workflows/publish-packages.yml:400]: Could you also check other operations (start, stop, restart) also if after the removal of the package `systemctl status hazelcast` reports that service is not known

**Review by ldziedziul — CHANGES_REQUESTED** [review]
Please add more systemctl tests

**Review by ldziedziul — APPROVED** [review]

**Comment by TomaszGaweda** [comment]
+1 from my side

**Pull request closed without merging**